test/Syntax/Parser/diags.swift

```diff
@@ -0,0 +1,27 @@
+// REQUIRES: syntax_parser_lib
+// RUN: %swift-syntax-parser-test %s -dump-diags | %FileCheck %s
+
+let number⁚ Int
+
+5 ‒ 5
+
+if (true ꝸꝸꝸ false) {}
+
+if (5 ‒ 5) == 0 {}
+
+// CHECK: Error: consecutive statements on a line must be separated by ';'
+// CHECK: offset: 117
+// CHECK: Fixit range: (117,117)
```

Comment:
Having the test utility print byte offsets is making it rather hard to write test cases where it is clear the output is correct, and making changes to the test cases. I'd suggest using `llvm::SourceMgr` to wrap the source text and then use `llvm::SourceMgr::getLineAndColumn` function to convert the offset to line/column and print it.

Then check for line/columns in the test case like `// CHECK: [[@LINE]]:10`, see example here: https://github.com/apple/swift/blob/f1d1df3b30046c6b9cfafe47215d42922c45fb57/test/Index/expressions.swift#L5